### DISCUSSION TITLE
MODBUS communication error

### DISCUSSION THREAD

**Discussion by elduderin07** [artisan-roaster-scope/artisan#697] [Q&A]
Hi there,

I am trying to connect my new Besca Bee roaster to Artisan but I keep getting "MODBUS communication error".

I have tried several solutions and workarounds with assistance from IT support of Besca but the problem persists and now the IT person informed me it might be an issue with my laptop or USB connection. This is the second laptop I have tried and faced the same problem.

Apparently the issue may lie with the fact that I am using a USB hub to connect my laptop/table hybrid (Microsoft Surface Go2 with USB-C input) to the roaster.

Any thoughts/ideas on how to address this or anyone facing similar issues?

**Comment by MAKOMO** [comment]
Did you try with a computer that features a regular (non USB-C) port with the roaster directly plugged-in, just to check if this USB-C HUB is the culprit?

This Artisan message indicates to me that Artisan can simply not reach out to the machine. You need to ensure to have the correct USB driver installed as required by the roasting machine (note that the FTDI driver is installed by default on most modern OS versions). Ask Besca for which one is needed. Then you need to select the correct serial port on configuration in Artisan. It should be the one added to the ports menu when you have the machine plugged in (just check for the difference in entries in plugged vs unplugged state). Then you should be ready to go. If there is no additional serial port entry in the ports popup on Artisan, this indicates that your operating system did not recognize the roaster which might be due to a bad connection or the lack of the correct driver.

**Reply by elduderin07** [reply]
Thanks for your response.
I have tried two other laptops with a regular USB port and Windows 10 - the one had the same issue (modbus communication error) and the other one which is older connected.
Can’t figure out what the issue is.
I installed the port drivers and nothing worked again

**Reply by elduderin07** [reply]
I’m wondering also if it’s a windows10 issue or a usb hub

**Reply by elduderin07** [reply]
Do you have any suggestions for drivers other than the ones recommended by artisan?

**Reply by MAKOMO** [reply]
Artisan works fine on Windows 8 as well as Windows 10. Regarding the required driver you need to ask Besca. I do not even know which communication hardware they build in their machines. The driver needs to fit the hardware in the roaster and need to work on your operating system. Nothing we can do from the Artisan side for you. Sorry.

**Reply by poundy** [reply]
my gut feel with a modbus error is that you have selected the wrong COM port. Just to be triply sure, open device manager with the roaster DISCONNECTED. Look at the COM PORT section, if it is there (it may not be if you have no other serial devices connected). If it is there, confirm what COM ports are in use; you will not connect to those in Artisan.

Then, connect the roaster and observe what devices show up, and confirm what COM port(s) appear. Take a screenshot and post here as well. If only one COM port appears and doesn't have a driver issue, go back to Artisan and make sure you have that selected in the setup, and make sure you have set all the other parameters as per Besca's instructions, and then try again and see if you still get the error.

**Reply by elduderin07** [reply]
There is only one COM port (COM4) and we have updated the drivers together with Besca's IT support. still didnt work...
My laptop has a USB-C input connection (with Windows 10) so I am wondering whether that is the issue

**Reply by poundy** [reply]
My laptop is a Surface Book 2, and it has a USB-A and USB-C but TBH I have only ever used the USB-A connector as I'm connected to a Bluetooth dongle to wirelessly connect to the roaster. I've been running Win10 no problem, and never had an issue connecting devices. I am running on the Win11 preview early builds, also no issues with communications to my roaster.

So on one of the devices you have tried, everything just worked. That's great. Means your configuration on that PC was working fine. That also means that fundamentally, Artisan is talking to the roaster correctly as well.  Unfortunately this does mean that there is something wrong at a hardware level with the combination of things you're trying to get working. Serial ports are so simple, there's no chance this is an OS issue, assuming you applied the correct and same driver in the working and non-working PCs. It also means that if you're using the same cables in both cases, there should be no issue in them either.  But if you've changed anything between those cases, you're closer to identifying what is not working. Replace those and see how that goes - if no change, you unfortunately can't assume that this means the first was working fine.

**Comment by jsphvrghs** [comment]
A little late here, but FWIW, I also have a USB-C port on my laptop and tried connecting my MODBUS to it. It did not work.
After a little digging, I noticed that the MODBUS to RS-485 converter LEDs were not lighting up. Turns out it requires power from the USB connector and the USB-C port on my laptop doesn't provide power. On the same laptop USB-A worked fine because the USB-A port transfers data AND provides power.
